HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,221 @@
+import pickle
+import string
+import re
+import urllib2
+import itertools
+import math as ma
+import os
+import pyspark
+
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+
+def link_string_to_KV(s):
+    src, dests = s.split(': ')
+    dests = [int(to) for to in dests.split(' ')]
+    return (int(src), dests)
+
+
+def fun(i):
+    return lambda x: (x[0], i + 1)
+
+
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+
+def find_distance(node, dist):
+        for i in dist:

```

Comment:
would be a little bit clearer if you had comments describing what exactly node and dist are. Is dist a dict? A list? Or, having the variable names be longer and more descriptive would be better.